xpallets/assets/src/lib.rs

```diff
@@ -237,41 +158,172 @@ decl_module! {
             Ok(())
         }
 
-        #[weight = <T as Config>::WeightInfo::set_asset_limit()]
-        pub fn set_asset_limit(origin, #[compact] id: AssetId, restrictions: AssetRestrictions) -> DispatchResult {
+        /// asset restriction method reserved for root
+        #[pallet::weight(<T as Config>::WeightInfo::zero())]
```

Comment:
```suggestion
        #[pallet::weight(<T as Config>::WeightInfo::set_asset_limit())]
```
Is there any other reason to set it to 0？